src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_SIZE = 10;
+    private int size;
+
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[])(new Object[DEFAULT_ARRAYLIST_SIZE]);
+    }
+
     @Override
     public void add(T value) {
-
+        if (size + 1 > array.length) {
+            growArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        outOfRangeCheck((index == size && size != 0) ? (index - 1) : index);
+        if (size + 1 > array.length) {
+            growArray();
+        }
+        System.arraycopy(array, index, array, index + 1,array.length - index - 1);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (size + list.size() > array.length) {
+            growArray();
+        }
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
```suggestion
        for (int i = 0; i < list.size(); i++) {
```
method add would increase the capacity if that would be needed